crypt-des-obsolete.c

```diff
@@ -49,7 +52,12 @@
 #include "crypt-obsolete.h"
 #include "alg-des.h"
 
-#if INCLUDE_encrypt || INCLUDE_encrypt_r || INCLUDE_setkey || INCLUDE_setkey_r
+#if ENABLE_OBSOLETE_API_ENOSYS
+#include <errno.h>
+#endif
```

Comment:
Throughout: it seems like it would be more readable and less of a future trip hazard if we just include errno.h unconditionally in each file that might need it.